### PR TITLE
roachtest: revert pebble benchmark test name; add commentary

### PR DESCRIPTION
**Pull request by nicktrav**
The name of the roachtest is used by the Pebble `mkbench` command to
generate a well-known directory structure and relied-upon by the
javascript on the Pebble benchmarks webpage.

Revert to using the original name for now, so as not to confuse the
`mkbench` tool.

Release note: None

### CODE REVIEW

**Review by bananabrick — CHANGES_REQUESTED** [review]
Thanks for adding some comments.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72622#-:-MoAVJMv-OS3XCc4NVCj:b-ju23z0)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jbowens and @nicktrav)

___
*[pkg/cmd/roachtest/tests/pebble.go, line 127 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72622#-MoAVL_o779HOxQqZUkI:-MoAVL_o779HOxQqZUkJ:b-urab5x) ([raw file](https://github.com/cockroachdb/cockroach/blob/6acdee0b531c23c28d264b9e5f09abbc1c897b1c/pkg/cmd/roachtest/tests/pebble.go#L127)):*
> ```Go
> 				// structure that is relied-upon by the javascript on the Pebble
> 				// Benchmarks webpage.
> 				Name:    fmt.Sprintf("pebble/ycsb/size=%d", size),
> ```

This doesn't work. We have 4 tests, and all of them need distinct names. By only varying size, we can only get two distinct names. As we've seen from last nights benchmark run, the other directory name worked, I'm curious what the core problem is here.

**Review by nicktrav — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72622)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @bananabrick and @jbowens)

___
*[pkg/cmd/roachtest/tests/pebble.go, line 127 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72622#-MoAVL_o779HOxQqZUkI:-MoAVr5v102D0obW89cD:b-2zc0x3) ([raw file](https://github.com/cockroachdb/cockroach/blob/6acdee0b531c23c28d264b9e5f09abbc1c897b1c/pkg/cmd/roachtest/tests/pebble.go#L127)):*
<details><summary><i>Previously, bananabrick (Arjun Nair) wrote…</i></summary><blockquote>

This doesn't work. We have 4 tests, and all of them need distinct names. By only varying size, we can only get two distinct names. As we've seen from last nights benchmark run, the other directory name worked, I'm curious what the core problem is here.
</blockquote></details>

Ah, you're right. Can't have the name clash. Thanks for pointing that out.

In terms of the core problem ...

Here's a [link](https://github.com/cockroachdb/pebble/blob/447349ef50e67d8766e3c1f5bb14eb6de63afd65/docs/index.html#L69) to the HTML. You can see the names of the `data-key`s match the name of the directory structure present in S3. With the previous change, we started including a new directory  (log into the AWS console and compare recent directories with what we had previously in the `pebble-benchmarks` bucket).

The problem here isn't so much that the data _won't_ be generated (it is, as evidenced by the data showing up on the benchmarks page), it's that the directory structure we write to S3 has suddenly changed.

This increases the cognitive overhead of trying to understand how this (already complex) benchmark data pipeline fits together - i.e. TeamCity calls runner script, which filters on a specific tag, which runs the roachtest(s), which then calls `mkbench`, which writes to S3 with the directory structure found on disk, which is then picked up by the Javascript and placed in the corresponding `chart` HTML element using the appropriate `data-key` as the index.

This is only working because in `mkbench` [we allow path names to have _more_ than six parts](https://github.com/cockroachdb/pebble/blob/447349ef50e67d8766e3c1f5bb14eb6de63afd65/internal/mkbench/ycsb.go#L151-L166). However, we are effectively dropping this new `duration=%d` dimension when merging the data into the `data.js` file. This is confusing, and [violates what the comment in `mkbench` states](https://github.com/cockroachdb/pebble/blob/447349ef50e67d8766e3c1f5bb14eb6de63afd65/internal/mkbench/ycsb.go#L151) (e.g. directory structure of the form`$date/pebble/ycsb/$name/$run/$file`).

Here's my proposal: we tack on a suffix for the 10 minute tests. That way we preserve the directory structure for the "golden data" that is written to S3 for the benchmarks. There are no assumptions about what the test name looks like when running a roachtest locally, so that should not be a problem or violate any assumptions elsewhere.

Alternatively, we persist both the 10 and 90 minute run data to S3, and update the benchmarks page to take this into account. However, we've already had the conversation / realization that 10 minute runs don't really produce reliable results. Instead, a shorter run is only really useful for testing.

tldr: Nothing is broken, however, I think we can save our future selves some confusion by preserving the original directory structure in the S3 bucket.

**Review by bananabrick — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72622)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jbowens and @nicktrav)

___
*[pkg/cmd/roachtest/tests/pebble.go, line 127 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72622#-MoAVL_o779HOxQqZUkI:-MoAfH8l05vFeK_jX2gS:bf1ae6j) ([raw file](https://github.com/cockroachdb/cockroach/blob/6acdee0b531c23c28d264b9e5f09abbc1c897b1c/pkg/cmd/roachtest/tests/pebble.go#L127)):*
<details><summary><i>Previously, nicktrav (Nick Travers) wrote…</i></summary><blockquote>

Ah, you're right. Can't have the name clash. Thanks for pointing that out.

In terms of the core problem ...

Here's a [link](https://github.com/cockroachdb/pebble/blob/447349ef50e67d8766e3c1f5bb14eb6de63afd65/docs/index.html#L69) to the HTML. You can see the names of the `data-key`s match the name of the directory structure present in S3. With the previous change, we started including a new directory  (log into the AWS console and compare recent directories with what we had previously in the `pebble-benchmarks` bucket).

The problem here isn't so much that the data _won't_ be generated (it is, as evidenced by the data showing up on the benchmarks page), it's that the directory structure we write to S3 has suddenly changed.

This increases the cognitive overhead of trying to understand how this (already complex) benchmark data pipeline fits together - i.e. TeamCity calls runner script, which filters on a specific tag, which runs the roachtest(s), which then calls `mkbench`, which writes to S3 with the directory structure found on disk, which is then picked up by the Javascript and placed in the corresponding `chart` HTML element using the appropriate `data-key` as the index.

This is only working because in `mkbench` [we allow path names to have _more_ than six parts](https://github.com/cockroachdb/pebble/blob/447349ef50e67d8766e3c1f5bb14eb6de63afd65/internal/mkbench/ycsb.go#L151-L166). However, we are effectively dropping this new `duration=%d` dimension when merging the data into the `data.js` file. This is confusing, and [violates what the comment in `mkbench` states](https://github.com/cockroachdb/pebble/blob/447349ef50e67d8766e3c1f5bb14eb6de63afd65/internal/mkbench/ycsb.go#L151) (e.g. directory structure of the form`$date/pebble/ycsb/$name/$run/$file`).

Here's my proposal: we tack on a suffix for the 10 minute tests. That way we preserve the directory structure for the "golden data" that is written to S3 for the benchmarks. There are no assumptions about what the test name looks like when running a roachtest locally, so that should not be a problem or violate any assumptions elsewhere.

Alternatively, we persist both the 10 and 90 minute run data to S3, and update the benchmarks page to take this into account. However, we've already had the conversation / realization that 10 minute runs don't really produce reliable results. Instead, a shorter run is only really useful for testing.

tldr: Nothing is broken, however, I think we can save our future selves some confusion by preserving the original directory structure in the S3 bucket.
</blockquote></details>

Thanks for the comprehensive reply. Yea, adding a duration suffix only for the 10 minute tests, with an explanation of why, sounds good to me.

**Review by nicktrav — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72622)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @bananabrick and @jbowens)

___
*[pkg/cmd/roachtest/tests/pebble.go, line 127 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72622#-MoAVL_o779HOxQqZUkI:-MoAl7HRBDdH30OuUxiD:bm2rdnx) ([raw file](https://github.com/cockroachdb/cockroach/blob/6acdee0b531c23c28d264b9e5f09abbc1c897b1c/pkg/cmd/roachtest/tests/pebble.go#L127)):*
<details><summary><i>Previously, bananabrick (Arjun Nair) wrote…</i></summary><blockquote>

Thanks for the comprehensive reply. Yea, adding a duration suffix only for the 10 minute tests, with an explanation of why, sounds good to me.
</blockquote></details>

Perfect. Updated.

**Comment by nicktrav** [comment]
TFTR!
